### DISCUSSION TITLE
GPIO 37 different at version 4.1

### DISCUSSION THREAD

**Discussion by Alexander89** [mciantyre/teensy4-rs#117] [General]
I'm currently trying to use SPI1 on a Teensy 4.1.

```rust
    let pins = bsp::t41::into_pins(peripherals.iomuxc);
    // ...
    let mut spi4 = spi1_builder.build(pins.p35, pins.p34, pins.p37);
```

After checking the docu https://www.pjrc.com/teensy/schematic41.png and https://www.pjrc.com/store/teensy41_card11a_rev3.png , it seems that port 37 is different, and B0_00 is not the pin to go on version 4.1 (it's B1_03)

I have no Idea how to fix it :-(

**Comment by mciantyre** [comment]
Ahh, just noticed the key piece of information: you're using a Teensy **4.1**. My test assumed a Teensy **4.0**. Thanks for sharing the

```rust
let pins = bsp::t41::into_pins(peripherals.iomuxc);
```

context. I can reproduce #115 issue when using the 4.1 pins.

You're correct; none of pins 34, 35, or 37 on the **4.1** are usable with an LPSPI peripheral. Could you select different pins? Or is the hardware design fixed?

If you must use those three pins, you could try using the FLEXIO peripheral to emulate SPI. Those three pins seem to support FLEXIO functions. See chapter 50 and section 50.4.3 from the reference manual. The introduction hints

> The FlexIO module is capable of supporting a wide range of protocols including, but not limited to: [...] SPI

I've not seen a Rust-based driver for the FLEXIO peripheral.

**Reply by Alexander89** [reply]
> just noticed the key piece of information: you're using a Teensy 4.1

I didn't notice the difference between his bords first. I thought 4.1 only extends the pins of 4.0.

I'm planning to use SPI3 now and opened a PR here: https://github.com/imxrt-rs/imxrt-iomuxc/pull/12 according to the IMXRT1060RM specs, I added all possible SPI "Pins"

If there are issues, I probably should do the FLEXIO workaround. I planned to do a useful SD integration first and don't get side traped again :see_no_evil:

**Reply by mciantyre** [reply]
Sounds great. Reach out if you hit other issues.

Not sure what you're planning to do with SPI, but keep in mind that imxrt-rs/imxrt-hal#111 is still open, and might affect your SPI behavior.

**Reply by Alexander89** [reply]
Wuhu :tada: 
After the update to `imxrt-iomuxc = "0.1.5"`, `let spi3 = spi3_builder.build(pins.p26, pins.p1, pins.p27);` is working as expected :tada: 

Thanks for your great and super short response time :-)
